src/Controller/Admin/LoginController.php

```diff
@@ -114,7 +114,10 @@ public function loginAction(
             return new RedirectResponse($redirectUrl);
         }
 
-        $csrfProtection->regenerateCsrfToken($request->getSession());
+        // check csrf token before generating a new one with force=true
+        if (!$csrfProtection->getCsrfToken($request->getSession())) {
```

Comment:
Thank you for the PR!
LGTM just maybe
```suggestion
        if (!$csrfProtection->getCsrfToken($request->getSession()) && is_null($request->get('deeplink'))) {
```
as i am not sure it should change the behaviour other than for  the deeplink

Reply:
@kingjia90 but some users might have bookmarked the `/admin/login` link. So, if they are simultaneously working in the backend in one tab and open another tab via the bookmark, the first tab will throw an Access Denied Exception, which is not the desired outcome.

Reply:
mhh you're right, that could be a case